scalus-examples/jvm/src/main/scala/scalus/examples/Betting.scala

```diff
@@ -0,0 +1,101 @@
+package scalus.examples
+
+import scalus.*
+import scalus.ledger.api.v3.*
+import scalus.ledger.api.v1.Address
+import scalus.prelude.*
+import scalus.prelude.Option.*
+import scalus.ledger.api.v2.OutputDatum
+import scalus.builtin.Data
+import scalus.builtin.Data.{FromData, ToData}
+import scalus.Compiler.compile
+import scalus.cardano.plutus.contract.blueprint.{Application, Blueprint}
+
+// https://github.com/cardano-foundation/cardano-template-and-ecosystem-monitoring/blob/main/bet/onchain/aiken/validators/bet.ak
+
+/** Represents the state of a two-player betting game The bet starts with player1 creating it, then
+  * player2 can join The oracle decides the winner and triggers the payout
+  */
+case class BetDatum(
+    /** The public key hash of the first player (bet creator) */
+    player1: PubKeyHash,
+    /** The public key hash of the second player (None if no one has joined yet) */
+    player2: PubKeyHash,
+    /** The public key hash of the trusted oracle who will announce the winner */
+    oracle: PubKeyHash,
+    /** The expiration time of the bet (in seconds since the epoch) */
+    expiration: PosixTime
+) derives FromData,
```

Comment:
In order `derives FromData, ToData` to work you need to add 
```scala
@Compile
object BetDatum
```
same for other types.